force-app/main/default/classes/LeadTriggerHandler.cls

```diff
@@ -104,12 +104,15 @@ public with sharing class LeadTriggerHandler {
 	 * - One of the errors is map related. Make sure you are using the correct contact map key
 	 */
 	public static void handleLeadAutoConvert(List<Lead> leads) {
-		// Step 1: Gather all lead emails
-        Map<Id,String> leadToEmailMap = new Map<Id,String>();
+		// Step 1: Gather all lead emails for non-converted leads
+		Map<Id, String> leadToEmailMap = new Map<Id, String>();
 		for (Lead lead : leads) {
-			leadToEmailMap.put(lead.Id, lead.Email);
+			// Check if the lead is already converted or if the email is null
+			if (!lead.IsConverted && lead.Email != null) {
+				leadToEmailMap.put(lead.Id, lead.Email);
+			}
 		}
-
+	
 		// Step 2: Find matching contacts based on email
 		Map<String, Contact> emailToContactMap = new Map<String, Contact>();
 		for (Contact c : [SELECT Id, Email, AccountId FROM Contact WHERE Email IN :leadToEmailMap.values()]) {
```

Comment:
I know it was probably sent to you that way but try to replace other developers single letter variables with meaningful variables